velox/functions/prestosql/types/JsonType.cpp

```diff
@@ -1288,6 +1289,17 @@ class JsonTypeFactories : public CustomTypeFactories {
   exec::CastOperatorPtr getCastOperator() const override {
     return std::make_shared<JsonCastOperator>();
   }
+
+  AbstractInputGeneratorPtr getInputGenerator(
+      const InputGeneratorConfig& config) const override {
+    return std::make_shared<fuzzer::JsonInputGenerator>(
+        config.seed_,
+        JSON(),
+        config.nullRatio_,
+        fuzzer::getRandomInputGenerator(
```

Comment:
This is the cause of the missing symbol. The link to velox_fuzzer... is removed by the monolithic wrapper function. We could link this explicitly but to be honest I don't think the type defintion is the right place for an input generator?

Reply:
Hi @assignUser, every custom type needs a logic to generate valid data, so the custom types and input generators are naturally bounded. So we add a new API getInputGenerator() to CustomTypeFactories to remind every custom type author to define the data generation logic intentionally. I'll try to move ConstrainedInputGenerators.h/cpp to another place not under fuzzer directories. Thank you for helping pinpoint the problem!

Reply:
I'm moving ConstrainedInputGenerators.h/cpp to velox/common/fuzzer: https://github.com/facebookincubator/velox/pull/12080.